cinema/views.py

```diff
@@ -103,6 +107,31 @@ def get_serializer_class(self):
 
         return MovieSerializer
 
+    @action(detail=True, methods=["post"],
+            parser_classes=[MultiPartParser, FormParser])
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        if "image" not in request.data:
+            return Response(
+                {"detail": "No image provided"},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+        image = request.data["image"]
+        try:
+            img = PIL.Image.open(image)
+            img.verify()
+        except (IOError, SyntaxError):
+            return Response(
+                {"detail": "Uploaded file is not a valid image."},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+        movie.image = image
+        movie.save()
+        return Response({
+            "status": "image uploaded",
+            "image": movie.image.url
+        }, status=status.HTTP_200_OK)
```

Comment:
You can provide here just url_path="upload-image", and then in get_serializer_class use MovieImageSerializer if self.action will be upload-image . And in this func just check if serializer is valid will be ok )